src/charm.py

```diff
@@ -186,22 +191,45 @@ def _on_install(self, _) -> None:
         except (apt.InvalidSourceError, ValueError, apt.GPGKeyError, URLError):
             self.unit.status = BlockedStatus("couldn't install MongoDB")
 
+        # Construct the mongod startup commandline args for systemd
+        machine_ip = self._unit_ip(self.unit)
+        mongod_start_args = " ".join(
```

Comment:
Considering that the idea is allow users to pass custom config file in the future, will you be parsing the file and adding to this list or writing the file directly to the fs? 
In the latter case, will you completely dismiss the `mongo_start_args`? (because, cmd line args will superseed config file config, in case on conflicts)

Reply:
I am not sure what is the best option here, perhaps @delgod knows?

Reply:
Awesome! This is a great explaination @delgod thanks for making this clear for me :)